File: codebase/e04-mvx/src/main/java/de/throsenheim/gui/locations/LocationModel.java
package de.throsenheim.gui.locations;

import javafx.beans.property.*;

import java.time.LocalDate;

/**
 * Model for locations
 *
 * @author dominik.haas
 */
public class LocationModel {

    private StringProperty id = new SimpleStringProperty();
    private StringProperty name = new SimpleStringProperty();

    private StringProperty label = new SimpleStringProperty();
    private StringProperty type = new SimpleStringProperty();
    private StringProperty subType = new SimpleStringProperty();
    private ObjectProperty<LocalDate> since = new SimpleObjectProperty<>();
    private BooleanProperty active = new SimpleBooleanProperty();

    private StringProperty street = new SimpleStringProperty();
    private StringProperty streetNumber = new SimpleStringProperty();

    private StringProperty zip = new SimpleStringProperty();
    private StringProperty city = new SimpleStringProperty();

    private StringProperty notes = new SimpleStringProperty();

    /**
     *  Constructor for events - this initialies the immutable id
     */
    public LocationModel() {
        id.set(java.util.UUID.randomUUID().toString());
    }

    public String getId() {
        return id.get();
    }

    public StringProperty idProperty() {
        return id;
    }

    public void setId(String id) {
        this.id.set(id);
    }

    public String getName() {
        return name.get();
    }

    public StringProperty nameProperty() {
        return name;
    }

    public void setName(String name) {
        this.name.set(name);
    }

    public String getLabel() {
        return label.get();
    }

    public StringProperty labelProperty() {
        return label;
    }

    public void setLabel(String label) {
        this.label.set(label);
    }

    public String getType() {
        return type.get();
    }

    public StringProperty typeProperty() {
        return type;
    }

    public void setType(String type) {
        this.type.set(type);
    }

    public String getSubType() {
        return subType.get();
    }

    public StringProperty subTypeProperty() {
        return subType;
    }

    public void setSubType(String subType) {
        this.subType.set(subType);
    }

    public LocalDate getSince() {
        return since.get();
    }

    public ObjectProperty<LocalDate> sinceProperty() {
        return since;
    }

    public void setSince(LocalDate since) {
        this.since.set(since);
    }

    public boolean isActive() {
        return active.get();
    }

    public BooleanProperty activeProperty() {
        return active;
    }

    public void setActive(boolean active) {
        this.active.set(active);
    }

    public String getStreet() {
        return street.get();
    }

    public StringProperty streetProperty() {
        return street;
    }

    public void setStreet(String street) {
        this.street.set(street);
    }

    public String getStreetNumber() {
        return streetNumber.get();
    }

    public StringProperty streetNumberProperty() {
        return streetNumber;
    }

    public void setStreetNumber(String streetNumber) {
        this.streetNumber.set(streetNumber);
    }

    public String getZip() {
        return zip.get();
    }

    public StringProperty zipProperty() {
        return zip;
    }

    public void setZip(String zip) {
        this.zip.set(zip);
    }

    public String getCity() {
        return city.get();
    }

    public StringProperty cityProperty() {
        return city;
    }

    public void setCity(String city) {
        this.city.set(city);
    }

    public String getNotes() {
        return notes.get();
    }

    public StringProperty notesProperty() {
        return notes;
    }

    public void setNotes(String notes) {
        this.notes.set(notes);
    }

    @Override
    public String toString() {
        return "LocationModel{" +
                "id=" + id +
                ", name=" + name +
                ", label=" + label +
                ", type=" + type +
                ", subType=" + subType +
                ", since=" + since +
                ", active=" + active +
                ", street=" + street +
                ", streetNumber=" + streetNumber +
                ", zip=" + zip +
                ", city=" + city +
                ", notes=" + notes +
                '}';
    }
}

File: codebase/e11-web2/src/LocationModel.ts
/**
 * Created by dominik.haas on 21.06.18.
 */
export class LocationModel {
  private _id?: string;
  private _name?: string;

  private _label?: string;
  private _type?: string;
  private _subType?: string;
  private _since?: Date;
  private _active?: boolean;

  private _street?: string;
  private _streetNumber?: string;
  private _zip?: string;
  private _city?: string;
  private _notes?: string;

  get id(): string | undefined {
    return this._id;
  }

  set id(value: string | undefined) {
    this._id = value;
  }

  get name(): string | undefined {
    return this._name;
  }

  set name(value: string | undefined) {
    this._name = value;
  }

  get label(): string | undefined {
    return this._label;
  }

  set label(value: string | undefined) {
    this._label = value;
  }

  get type(): string | undefined {
    return this._type;
  }

  set type(value: string | undefined) {
    this._type = value;
  }

  get subType(): string | undefined {
    return this._subType;
  }

  set subType(value: string | undefined) {
    this._subType = value;
  }

  get since(): Date | undefined {
    return this._since;
  }

  set since(value: Date | undefined) {
    this._since = value;
  }

  get active(): boolean | undefined {
    return this._active;
  }

  set active(value: boolean | undefined) {
    this._active = value;
  }

  get street(): string | undefined {
    return this._street;
  }

  set street(value: string | undefined) {
    this._street = value;
  }

  get streetNumber(): string | undefined {
    return this._streetNumber;
  }

  set streetNumber(value: string | undefined) {
    this._streetNumber = value;
  }

  get zip(): string | undefined {
    return this._zip;
  }

  set zip(value: string | undefined) {
    this._zip = value;
  }

  get city(): string | undefined {
    return this._city;
  }

  set city(value: string | undefined) {
    this._city = value;
  }

  get notes(): string | undefined {
    return this._notes;
  }

  set notes(value: string | undefined) {
    this._notes = value;
  }

  toString(): string {
    return JSON.stringify(this);
  }
}
